src/Microsoft.Extensions.Configuration.UserSecrets/ConfigurationExtensions.cs

```diff
@@ -76,15 +24,15 @@ public static IConfigurationBuilder AddUserSecrets(this IConfigurationBuilder co
                 throw new ArgumentNullException(nameof(configuration));
             }
 
-            var projectPath = configuration.GetProjectPath();
-            if (string.IsNullOrEmpty(projectPath))
+            var fileProvider = configuration.GetFileProvider();
+            if (fileProvider == null)
             {
                 throw new InvalidOperationException(Resources.FormatError_MissingBasePath(
-                        projectPath,
                         typeof(IConfigurationBuilder).Name,
-                        ProjectPathKey));
+                        "FileProvider"));
```

Comment:
Hmmm, you are actually dealing with the extension method and that doesn't change unless we change it. Regardless, it is ok to have this check. It will just not happen very often.